src/KafkaW/ProducerTopic.cpp

```diff
@@ -44,59 +26,67 @@ ProducerTopic::ProducerTopic(ProducerTopic &&x) noexcept {
   std::swap(DoCopyMsg, x.DoCopyMsg);
 }
 
-struct Msg_ : public Producer::Msg {
-  vector<unsigned char> v;
+struct Msg_ : public ProducerMessage {
+  std::vector<unsigned char> v;
```

Comment:
It is not a problem as such but I do find it slightly weird that you are using a `std:vector` here when there are several other (in my opinion) more obvious options.